client/src/utils/similarity.js

```diff
@@ -72,32 +70,48 @@ export default class Similarity {
     /* calculates similarity based on difference between values
     in same position 
     range from 0 - 100; closer to 0 means more similar */
+    a = this.convertObjectToVector(a)
+    b = this.convertObjectToVector(b)
     let differenceSum = 0
-    let scaleSum = 0
     /* set up scale values 
-    0 - acousticness
-    1 - danceability
-    2 - energy
-    3 - instrumentalness
-    4 - key
-    5 - liveness
-    6 - loudness
-    7 - mode
-    8 - speechiness
-    9 - time_signature
-    10 - valence
+    acousticness
+    danceability
+    energy
+    instrumentalness
+    key
+    liveness
+    loudness
+    mode
+    speechiness
+    time_signature
+    valence
     */
     let maxValue = 0
-    const scaleValues = [1.5, 1.8, 2, 1.5, .5, .3, .5, .5, 1.5, .5, 2]
+    const scaleValueObject = {
```

Comment:
This should probably be a const of the class since it won't change for each method call.